modules/vactory_redirect/src/Form/UploadUrlsCsv.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+namespace Drupal\vactory_redirect\Form;
+
+use Drupal\Core\Form\ConfigFormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\File\FileSystemInterface;
+use Drupal\Core\File\Exception\InvalidStreamWrapperException;
+use Drupal\Core\File\Exception\DirectoryNotReadyException;
+
+/**
+ * Configure vactory_redirect settings for this site.
+ */
+class UploadUrlsCsv extends ConfigFormBase {
+
+  const SETTINGS = 'vactory_redirect.settings';
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'vactory_redirect';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getEditableConfigNames() {
+    return [
+      static::SETTINGS,
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    return parent::buildForm($form, $form_state);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state) {
+    $uri = "redirections/urls.csv";
+    $destination = "private://redirections/" ;
+    if (file_exists($uri)) {
```

Comment:
@meryemdibe 

- The content of uploaded CSV file should override `redirections/urls.csv` file
- The destination directory `private://redirections/` should be created automatically in case it doesn't already exist.